### PR TITLE
mmseqs2: init at 14-7e284

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

MMseqs2 (Many-against-Many sequence searching) is a software suite to search and cluster huge protein and nucleotide sequence sets.

https://github.com/soedinglab/mmseqs2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/mmseqs2/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, xxd
+, perl
+, enableAvx2 ? stdenv.hostPlatform.avx2Support
+, enableSse4_1 ? stdenv.hostPlatform.sse4_1Support
+, enableMpi ? (false && !stdenv.isDarwin)
```
Comment (viraptor) [inline pkgs/applications/science/biology/mmseqs2/default.nix:9]: What's the idea here? The default will always evaluate to false.

File: pkgs/applications/science/biology/mmseqs2/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+, mpi
+, openmp
+, zlib
+, bzip2
+}:
+
+with lib;
```
Comment (viraptor) [inline pkgs/applications/science/biology/mmseqs2/default.nix:16]: That's pretty uncommon. Could we have just `lib.optional` where necessary? It's not a big distraction to write it out in full.

File: pkgs/applications/science/biology/mmseqs2/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/applications/science/biology/mmseqs2/default.nix:9]: I wanted it to be false when the target is darwin, but I should use the variable as "enablaMpi && !stdenv.isDarwin" when calling.
Also, I misunderstood that MPI is not available in darwin, so I will change it to merely false.

File: pkgs/applications/science/biology/mmseqs2/default.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/applications/science/biology/mmseqs2/default.nix:16]: I followed the nixos/modules' code, but it is very few in pkgs. I will change it.
Thank you!

**Comment by viraptor** [comment]
Just tried this and there's an extra file installed for completions (`util/bash-completion.sh`) which should be handled in a better way (search for `installShellCompletion` for examples). You could delete / recreate it properly in postInstall.

**Comment by natsukium** [comment]
Thank you for your suggestion.
I added postInstall phase.

**Pull request merged**